nodejs/week3/homework/warmup/index.js

```diff
@@ -1,6 +1,52 @@
 const express = require("express");
+const bodyParser = require("body-parser");
 const app = express();
 
+app.use(
+  bodyParser.urlencoded({
+    extended: true,
+  })
+);
 app.get("/", (req, res) => res.send("nodejs week3 homework"));
 
+app.get("/calculator/add", (req, res) => {
+  const params = req.query;
+  console.log(params);
+  const valuecs = Object.values(params);
+  let result = values.reduce((a, b) => parseInt(a) + parseInt(b));
+  res.send(`${result}`);
+});
+
+app.get("/calculator/multiply", (req, res) => {
+  const params = req.query;
+  console.log(params);
+  const values = Object.values(params);
+  let result = values.reduce((a, b) => parseInt(a) * parseInt(b));
+  res.send(`${result}`);
+});
+app.get("/numbers/subtraction", (req, res) => {
+  const params = req.query;
+  console.log(params);
+  res.send(`${parseInt(params.first) - parseInt(params.second)}`);
+});
+app.get("/calculator/division", (req, res) => {
+  const params = req.query;
+  console.log(params);
+  res.send(`${parseInt(params.first) / parseInt(params.second)}`);
+});
+app.post("/calculator/multiply", (req, res) => {
+  const query = req.body;
+  const keys = Object.values(query);
+
+  console.log(query);
+  console.log(keys);
+  const result = keys.reduce((a, b) => parseInt(a) * parseInt(b));
```

Comment:
An idea to avoid copy pasting the same code, for instance
```
const result = keys.reduce((a, b) => parseInt(a) * parseInt(b));
```
would be do to create a separate function, eg multiplyNumbers, and call that one both in the GET and the POST request.
That way, it's cleaner, and if say you want to add an extra step in your multiplication, you will only have to change it in one place